build/webcomponents/js/field-category/field-category.js

```diff
@@ -0,0 +1,35 @@
+customElements.define('joomla-field-category', class extends HTMLElement {
+  constructor() {
+    super();
+
+    this.element = '';
+    Joomla.loadingLayer('load', document.body);
+  }
+
+  connectedCallback() {
+    // Check if custom fields are enabled
+    if (this.getAttribute('custom-fields-enabled') !== 'true') {
+      return;
+    }
+
+    this.element = this.querySelector('select');
+    this.categoryHasChanged = this.categoryHasChanged.bind(this);
+
+    if (!this.element.value !== this.getAttribute('custom-fields-cat-id')) {
```

Comment:
why are we pefixing this with "custom-fields-xxx" why not just "category-id"?

Reply:
I guess we can do that